pkg/client/unversioned/discovery_client.go

```diff
@@ -71,27 +69,18 @@ func apiVersionsToAPIGroup(apiVersions *unversioned.APIVersions) (apiGroup unver
 	return
 }
 
-func (d *DiscoveryClient) get(url string) (resp *http.Response, err error) {
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
-	}
-	return d.httpClient.Do(req)
-}
-
 // ServerGroups returns the supported groups, with information like supported versions and the
 // preferred version.
 func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList, err error) {
 	// Get the groupVersions exposed at /api
-	url := d.baseURL
+	url := url.URL{}
 	url.Path = "/api"
-	resp, err := d.get(url.String())
+	resp, err := d.Get().AbsPath(url.String()).Do().Raw()
```

Comment:
I think part of the point of clayton's request is that you should use `err := ... .Do().Into(&v)`, which may or may not involve fixing up the codec the RESTClient is using.